### ISSUE TITLE
Move the jenkins-stats repositories to the jenkins-infra organization

### ISSUE THREAD

**Issue by gounthar** [jenkins-infra/helpdesk#4017]
### Service(s)

Other

### Summary

@jmMeessen has developed a set of tools to provide the community with insights into its contributors. Now that these tools have proven effective and stable, we would like them to become part of the `jenkins-infra` organization.

The repositories for these tools are:

[jenkins-top-submitters](https://github.com/jmMeessen/jenkins-top-submitters)
[jenkins-submitter-stats](https://github.com/jmMeessen/jenkins-submitter-stats)
The corresponding `Homebrew` tap can be found at [jmMeessen/homebrew-tap](https://github.com/jmMeessen/homebrew-tap).

The statistics, including scripts and data, are stored at [jmMeessen/jenkins-stats](https://github.com/jmMeessen/jenkins-stats).

What are the prerequisites for moving these repositories, including their issues, into the `jenkins-infra` organization?

Thank you for your assistance.

**Comment by jmMeessen** [comment]
Little correction/additional info to the above:
- the main stats extraction application (bash based) is living in [jenkins-submitter-stats](https://github.com/jmMeessen/jenkins-submitter-stats). The monthly extracted data (jenkins and benchmarked projects) is stored in the same GitHub repository.
- Ad-hoc tooling, used by the above process, is living in [jmMeessen/jenkins-stats](https://github.com/jmMeessen/jenkins-stats) and [jenkins-top-submitters](https://github.com/jmMeessen/jenkins-top-submitters). These tools are distributed via a homebrew "repository" located at [jmMeessen/homebrew-tap](https://github.com/jmMeessen/homebrew-tap).

These four Git repositories should be migrated to the `jenkins-infra` organisation as stated by @gounthar . Additionally, a (dedicated ?) GitHub app is needed to update homebrew-tap (part of a GitHub Action based CD process).

**Comment by lemeurherve** [comment]
Sorry for the delay, I have planned a pairing session tomorrow afternoon with @gounthar and @jmMeessen to transfer these repositories.

**Comment by lemeurherve** [comment]
This and some adaptations will allow unblock https://github.com/jenkins-infra/contributor-spotlight/pull/141

**Comment by gounthar** [comment]
Thanks, @lemeurherve !

**Comment by lemeurherve** [comment]
The 3 main repositories have been imported:
- https://github.com/jmMeessen/jenkins-submitter-stats: https://github.com/jenkins-infra/jenkins-contribution-stats
- https://github.com/jmMeessen/jenkins-stats: https://github.com/jenkins-infra/jenkins-contribution-extractor
- https://github.com/jmMeessen/jenkins-top-submitters: https://github.com/jenkins-infra/jenkins-contribution-aggregator

@gounthar and @jmMeessen are now working on updating references to these new names, then the first two repositories have to be configured to get goreleaser publishings corresponding taps to https://github.com/jenkins-infra/homebrew-tap like what's done in https://github.com/jenkins-infra/jenkins-version/blob/1342d2771b99c6de6afee408d4b1916c1555cab3/.goreleaser.yml#L63-L76.

**Comment by jmMeessen** [comment]
The code coverage automation doesn't work anymore for the two GO tools. It requires at least an API token so that the coverage data can be uploaded. See https://docs.codecov.com/docs/adding-the-codecov-token

Should this be tracked here or do I open a new issue?

It is a handy feature especially as it fails the build if new code is added via a PR and is not covered by 80% coverage. A good reminder not to forget automated tests.

**Comment by lemeurherve** [comment]
I would open a dedicated issue for the installation of codecov.

**Comment by jmMeessen** [comment]
The migration is blocked because of the HomeBrew token issue #4141 . This should be solved before @jmMeessen retires and the next stats run (July 1st)

**Comment by jmMeessen** [comment]
> I would open a dedicated issue for the installation of codecov.

non-blocking issue has been created #4142

**Comment by jmMeessen** [comment]
Current summary:

This issue is blocked because of these two (time) critical issues:
- #4141 
- #4143 

The following issue is non-critical:
- #4142 

@lemeurherve, what is the current plan/schedule to solve these issues caused by the incomplete migration ?

cc @gounthar , @MarkEWaite

**Comment by lemeurherve** [comment]
The plan will be discussed during our team infra meeting this afternoon, I'll report it here after.

**Comment by dduportal** [comment]
> The plan will be discussed during our team infra meeting this afternoon, I'll report it here after.

For info, we added the sub-issues in the new infra milestone and have started working on it

**Comment by dduportal** [comment]
> Current summary:
> 
> This issue is blocked because of these two (time) critical issues:
> 
>     * [Release of `jenkins-contribution-*` GO applications fail: homebrew token not found #4141](https://github.com/jenkins-infra/helpdesk/issues/4141)
> 
>     * [[contribution stats] publication of honored contributor fails #4143](https://github.com/jenkins-infra/helpdesk/issues/4143)
> 
> 
> The following issue is non-critical:
> 
>     * [Compilation of `jenkins-contribution-*` GO applications does not publish CodeCoverage report #4142](https://github.com/jenkins-infra/helpdesk/issues/4142)
> 
> 
> @lemeurherve, what is the current plan/schedule to solve these issues caused by the incomplete migration ?
> 
> cc @gounthar , @MarkEWaite

As far as I can tell, the repositories have been migrated and the issues mentioned have been fixed.

@gounthar @jmMeessen @lemeurherve what are the remaning tasks on this topic for you?

**Comment by lemeurherve** [comment]
None, @jmMeessen confirmed all of them, this issue can be closed.

**Comment by jmMeessen** [comment]
I confirm: all lights green. We are not blocked anymore.

@gounthar and myself are going through the lasts steps to validate the migration (validation of the automatic processes of `jenkins-contribution-stats`). We do not anticipate further issues but leave us until Friday June 22 PM to confirm this definitively and close this issue.

**Comment by gounthar** [comment]
It looks like the migration has been successful.
Thanks a lot to all the persons involved.
